utils/config.mli

```diff
@@ -248,6 +248,7 @@ val native_dynlink: bool
 val afl_instrument : bool
 (** Whether afl-fuzz instrumentation is generated by default *)
 
+val ar_supports_diversion: bool
```

Comment:
Where does the term 'diversion' come from?  Windows and clang both refer to them as [response files](https://man.openbsd.org/ar.1#Other).  (This function could also use a brief ocamldoc)

Reply:
Good question: I just carried on the name from #9492 and https://github.com/ocaml/ocaml/blob/59a9b02ea1f28053297d6901689db892d8b828d6/utils/ccomp.ml#L41. Maybe @xavierleroy has a better answer about this term.

We could switch to `ar_supports_response_files` or even `ar_supports_at_file` if we want to avoid the "diversion" terminology.

Reply:
I can't remember exactly why I used "diversion" here.  It's sometimes used to refer to parts of a script being stored in a temporary file and replaced by the file name, see e.g. https://www.ibm.com/docs/en/zos/2.4.0?topic=input-text-diversion .  Maybe that's what I had in mind.  But "response file" is clearer.

Reply:
I switched `diversion` to `response_file`.